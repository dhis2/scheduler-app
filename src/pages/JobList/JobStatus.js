import React from 'react'
import { string } from 'prop-types'
import { Tag } from '@dhis2/ui-core'

const JobStatus = ({ status }) => {
    switch (status) {
        case 'STOPPED':
        case 'DISABLED':
            return <Tag>{status}</Tag>
        case 'RUNNING':
        case 'NOT_STARTED':
        case 'SCHEDULED':
            return <Tag neutral>{status}</Tag>
        case 'FAILED':
            return <Tag negative>{status}</Tag>
        case 'DONE':
            return <Tag positive>{status}</Tag>
        default:
            return null
    }
}

JobStatus.propTypes = {
    status: string.isRequired,
}

export default JobStatus
