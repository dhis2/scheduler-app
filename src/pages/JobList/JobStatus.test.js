import React from 'react'
import { shallow } from 'enzyme'
import JobStatus from './JobStatus'

const statuses = [
    'STOPPED',
    'DISABLED',
    'RUNNING',
    'NOT_STARTED',
    'SCHEDULED',
    'FAILED',
    'DONE',
]

describe('<JobStatus>', () => {
    it.each(statuses)('renders correctly for %s status', status => {
        const wrapper = shallow(<JobStatus status={status} />)
        expect(wrapper).toMatchSnapshot()
    })

    it('renders correctly for an invalid status', () => {
        const wrapper = shallow(<JobStatus status="INVALID" />)

        expect(wrapper).toMatchSnapshot()
    })
})
