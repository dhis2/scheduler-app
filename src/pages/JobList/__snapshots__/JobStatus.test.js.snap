// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<JobStatus> renders correctly for DISABLED status 1`] = `
<Tag
  dataTest="dhis2-uicore-tag"
>
  DISABLED
</Tag>
`;

exports[`<JobStatus> renders correctly for DONE status 1`] = `
<Tag
  dataTest="dhis2-uicore-tag"
  positive={true}
>
  DONE
</Tag>
`;

exports[`<JobStatus> renders correctly for FAILED status 1`] = `
<Tag
  dataTest="dhis2-uicore-tag"
  negative={true}
>
  FAILED
</Tag>
`;

exports[`<JobStatus> renders correctly for NOT_STARTED status 1`] = `
<Tag
  dataTest="dhis2-uicore-tag"
  neutral={true}
>
  NOT_STARTED
</Tag>
`;

exports[`<JobStatus> renders correctly for RUNNING status 1`] = `
<Tag
  dataTest="dhis2-uicore-tag"
  neutral={true}
>
  RUNNING
</Tag>
`;

exports[`<JobStatus> renders correctly for SCHEDULED status 1`] = `
<Tag
  dataTest="dhis2-uicore-tag"
  neutral={true}
>
  SCHEDULED
</Tag>
`;

exports[`<JobStatus> renders correctly for STOPPED status 1`] = `
<Tag
  dataTest="dhis2-uicore-tag"
>
  STOPPED
</Tag>
`;

exports[`<JobStatus> renders correctly for an invalid status 1`] = `""`;
